manifests/accesslogin.pp

```diff
@@ -1,18 +1,61 @@
-# == Class: pam::accesslogin
+# @summary Manage login access
+#   See PAM_ACCESS(8)
 #
-# Manage login access
+# @example Declaring the class
+#   include ::pam::accesslogin
 #
-# See PAM_ACCESS(8)
+# @param access_conf_path
+#   Path to access.conf.
+#
+# @param access_conf_owner
+#   Owner of access.conf.
+#
+# @param access_conf_group
+#   Group of access.conf.
+#
+# @param access_conf_mode
+#   Mode of access.conf.
+#
+# @param access_conf_template
+#   Content template of access.conf.
+#
+# @param allowed_users
+#   String, Array or Hash of strings and/or arrays to configure users and
+#   origins in access.conf. The default allows the root user/group from
+#   origin 'ALL'.
 #
 class pam::accesslogin (
-  Stdlib::Absolutepath $access_conf_path  = '/etc/security/access.conf',
-  String $access_conf_owner               = 'root',
-  String $access_conf_group               = 'root',
-  Stdlib::Filemode $access_conf_mode      = '0644',
-  String $access_conf_template            = 'pam/access.conf.erb',
-) {
+  Stdlib::Absolutepath $access_conf_path      = '/etc/security/access.conf',
+  String $access_conf_owner                   = 'root',
+  String $access_conf_group                   = 'root',
+  Stdlib::Filemode $access_conf_mode          = '0644',
+  String $access_conf_template                = 'pam/access.conf.erb',
+  Variant[Array, Hash, String] $allowed_users = $pam::allowed_users,
+) inherits ::pam {
```

Comment:
we should not be inheriting a class with parameters.

Reply:
Is there another difference when including a class via `inherits` or [`require`](https://github.com/ghoneycutt/puppet-module-pam/blob/master/manifests/accesslogin.pp#L15) beside the timing ?

- We could remove $allowed_users from main class and move it to accesslogin exclusively.

- The other way around would be something like this pseudo code:

```
class pam::accesslogin (
  Optional[Variant[Array, Hash, String]] $allowed_users = undef,
) {

case $allowed_users {
  undef: { $allowed_users_real = $pam::allowed_users }
  default: $allowed_users_real = $allowed_users }
}
```

Reply:
+1 to moving `allowed_users` from `pam` to `pam::accesslogin` because the variable is only used in access.conf which is managed by `pam::accesslogin`.

Reply:
fixed

Reply:
Having issues to test package_name in accesslogin as it also comes from main class pam that is not included anymore now. Usual technique to use pre_conditions didn't worked for me today :(